spec/controllers/tasks_controller_spec.rb

```diff
@@ -404,6 +404,48 @@
     end
   end
 
+  describe 'POST #duplicate' do
+    subject(:post_duplicate) { post :duplicate, params: {id: task.to_param} }
+
+    let(:group) { create(:group) }
+    let(:collection) { create(:collection) }
+
+    let!(:task) { create(:task, valid_attributes) }
+    let(:valid_attributes) do
+      {
+        user:,
+        title: 'title',
+        license: create(:license, name: 'license'),
+        groups: [group],
+        collections: [collection],
+      }
+    end
```

Comment:
I haven't tested the suggestion, but you might be able to leverage the `build_list` feature here:

```suggestion
    let(:groups) { create_list(:group, 1) }
    let(:collections) { create_list(:collection, 1) }

    let!(:task) { create(:task, valid_attributes) }
    let(:valid_attributes) do
      {
        user:,
        title: 'title',
        license: create(:license, name: 'license'),
        groups:,
        collections:,
      }
    end
```

Reply:
Yes, it does indeed work. I'm going to upload it alongside the other suggestions.